File: virtual-world-cpp/VirtualWorld/Plants/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(int x, int y, World *world) : Plant(x, y, 0, 0, guaranaCode, world) {}

Plant *Guarana::Clone(int x, int y, World *world) {
    return new Guarana(x, y, world);
}

std::string Guarana::GetName() {
    return "Guarana";
}

bool Guarana::AttackPaired(Organism *attacker) {
    attacker->setStrength(attacker->getStrength() + 3);
    this->world->AddMessage(attacker->GetName() + " ate guarana and gained 3 strength points!");
    return false;
}

File: virtual-world-java/src/main/java/me/knasuta/virtualworld/simulation/plants/Guarana.java
package me.knasuta.virtualworld.simulation.plants;

import me.knasuta.virtualworld.simulation.Organism;
import me.knasuta.virtualworld.simulation.Point;
import me.knasuta.virtualworld.simulation.World;

import java.awt.*;

public class Guarana extends Plant{
    @Override
    public boolean AttackPaired(Organism other) {
        other.setStrength(other.getStrength() + 3);
        world.AddMessage(other.getName() + " ate guarana and gained 3 strength");
        return false;
    }
    public Guarana(Point location, World world) {
        super(location, 0, 0, world);
    }
    @Override
    public String getName() {
        return "Guarana";
    }

    @Override
    public Organism Clone(Point location, World world) {
        return new Guarana(location, world);
    }

    @Override
    public Color getColor() {
        return new Color(255, 0, 255);
    }
}
